src/theory/bv/bv_solver_bitblast.cpp

```diff
@@ -408,6 +411,47 @@ void BVSolverBitblast::handleEagerAtom(TNode fact, bool assertFact)
   registeredAtoms.clear();
 }
 
+std::vector<Node> BVSolverBitblast::getProofNodes(proof::DratProof dratProof)
+{
+  NodeManager* nm = NodeManager::currentNM();
+  Node cl = nm->mkBoundVar("cl", nm->booleanType());
+  Node del = nm->mkBoundVar("del", nm->booleanType());
+  Node lastFalseResolution = nm->mkConst<bool>(false);
+
+  std::vector<Node> args;
+  for (const proof::DratInstruction instruction : dratProof.getInstructions())
+  {
+    if (instruction.d_clause.size() == 0 && instruction.d_clause[0].toInt() == 0)
+    {
+      args.push_back(nm->mkNode(kind::SEXPR, {cl, lastFalseResolution}));
+      break;
+    }
+    std::vector<Node> clauseNodes;
+    if (instruction.d_kind == proof::DratInstructionKind::DELETION)
+    {
+      clauseNodes.emplace_back(del);
+    }
+    else
+    {
+      clauseNodes.emplace_back(cl);
+    }
+    for (const prop::SatLiteral literal : instruction.d_clause)
+    {
+      Node fact = d_cnfStream->getNode(literal);
+      if (fact.isNull()) {
+        std::ostringstream errmsg;
+        errmsg << "Not found node corresponding to literal from drat proof: \""
+                << literal
+                << "\"";
+        throw Exception(errmsg.str());
```

Comment:
We try to avoid using exceptions. Instead add an assertion that `fact` must not be null.